clients/src/test/java/org/apache/kafka/common/network/NioEchoServer.java

```diff
@@ -90,7 +90,7 @@ public String metricNameSuffix() {
     private volatile boolean closeKafkaChannels;
     private final DelegationTokenCache tokenCache;
     private final Time time;
-    private final AtomicLong idGenerator = new AtomicLong();
+    private final AtomicInteger nextConnectionIndex = new AtomicInteger();
```

Comment:
nit: this makes wraparound worse, can you apply the explicit wraparound logic that SocketServer uses?

Reply:
Sure, I will make sure they have same logic.